projects/ngrx-auto-entity/src/lib/util.ts

```diff
@@ -541,12 +576,29 @@ export const buildFeatureState = <TState extends IEntityState<TModel>, TParentSt
   selectParentState: MemoizedSelector<object, TParentState>,
   extraInitialState?: any
 ): IModelState<TParentState, TState, TModel> => {
-  const modelName = camelCase(new type().constructor.name);
+  const instance = new type();
+  console.log(`NGRX-AE: Building entity for: ${type.name}; constructor: ${instance.constructor.name}`);
+  const opts = type[ENTITY_OPTS_PROP] || {
+    modelName: instance.constructor.name
```

Comment:
Will this be a problem while this bug is open? https://github.com/webpack-contrib/uglifyjs-webpack-plugin/issues/269

Reply:
This is actually why I made these changes as I came across that issue while researching the problems both mxdmedia and I have had. Previously, I was only using `instance.constructor.name`. What I've done is add the ability to override the name of the model using  the new @Entity decorator, which attaches the ENTITY_OPTS_PROP to the type itself. That way, when the constructor.name gets mangled, your explicitly specified name will still work. This has been tested with prod builds and it works. I kept the constructor.name here as a fallback for now.

Reply:
Gotcha, makes sense.